[docs] Add documentation on auto translation (#131)
OpenBAS-Platform/docs: docs/development/translations.md

# Adding translations

### Introduction

> This guide explains how to add translations when developing on OpenBAS. We have 3 files to support the app’s languages: en.json (the reference file - for english), fr.son (for french) and zh.json (for chinese). We have set up some elements to ease the translation process.

Reviewer: ```suggestion
This guide explains how to add translations when developing with OpenBAS. We have 3 files to support the app’s languages: en.json (the reference file - for english), fr.son (for french) and zh.json (for chinese). We have set up some elements to streamline the translation process.
```

#### 1. Using scripts

> You can find translations management scripts in the package.json file.

Reviewer 2: The format list is not taking into account.
![image](https://github.com/user-attachments/assets/aaa2f7b2-00ff-424f-8350-d74206c7dae4)

- extract-translation : will fetch all the files and add new translations keys in en.json 
- auto-translation:fr : will translate all the elements of the en.json in french and add them in the fr.json
- auto-translation:cn : will translate all the elements of the en.json in chinese and add them in the zh.json

> - auto-translation: all : will translate all the elements of the en.json in all the app's supoorted languages and add them in the respective files

Author: The **all** is on purpose, that script translates in in french and chinese automatically
